File: spring-cloud-integrated-demo/user-service/user-service-app/src/main/java/com/github/jccode/springcloud/integrateddemo/user/service/UserService.java
package com.github.jccode.springcloud.integrateddemo.user.service;

import com.github.jccode.springbootsample.core.repo.CrudMapper;
import com.github.jccode.springbootsample.core.service.CrudServiceImpl;
import com.github.jccode.springcloud.integrateddemo.user.model.User;
import com.github.jccode.springcloud.integrateddemo.user.model.UserCriteria;
import com.github.jccode.springcloud.integrateddemo.user.repo.UserMapper;
import com.google.common.base.Preconditions;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService extends CrudServiceImpl<User> {

    @Autowired
    private UserMapper userMapper;

    @Autowired
    public UserService(CrudMapper<User> mapper) {
        super(mapper);
    }

    public List<User> findByName(String name) {
        Preconditions.checkNotNull(name);
        UserCriteria criteria = new UserCriteria();
        criteria.createCriteria().andNameEqualTo(name);
        return userMapper.selectByExample(criteria);
    }
}

File: spring-cloud-integrated-demo/user-service/user-service-app/src/main/scala/com/github/jccode/springcloud/integrateddemo/user/service/UserService.scala
package com.github.jccode.springcloud.integrateddemo.user.service

import com.github.jccode.springbootsample.core.service.CrudServiceImpl
import com.github.jccode.springcloud.integrateddemo.user.model.{User, UserCriteria}
import com.github.jccode.springcloud.integrateddemo.user.repo.UserMapper
import com.google.common.base.Preconditions
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import scala.collection.JavaConverters._

@Service
class UserService(@Autowired override val mapper: UserMapper) extends CrudServiceImpl[User](mapper) {

  def findByName(name: String): List[User] = {
    Preconditions.checkNotNull(name)
    val criteria = new UserCriteria
    criteria.createCriteria().andNameEqualTo(name)
    mapper.selectByExample(criteria).asScala.toList
  }

  def findAll: List[User] = {
    mapper.selectByExample(new UserCriteria).asScala.toList
  }
}
